test/Chainweb/Test/MultiNode.hs

```diff
@@ -265,49 +273,88 @@ runNodes loglevel write stateVar baseConf n rdb = do
             | otherwise ->
                 setBootstrapPeerInfo <$> readMVar bootstrapPortVar <*> pure baseConf
 
-        multiNode loglevel write stateVar bootstrapPortVar conf rdb i
+        multiNode loglevel write bootstrapPortVar conf rdb pactDbDir i inner
 
 runNodesForSeconds
     :: LogLevel
         -- ^ Loglevel
+    -> (T.Text -> IO ())
+        -- ^ logging backend callback
     -> ChainwebConfiguration
-    -> ChainwebVersion
     -> Natural
         -- ^ Number of chainweb consensus nodes
     -> Seconds
         -- ^ test duration in seconds
-    -> (T.Text -> IO ())
-        -- ^ logging backend callback
     -> RocksDb
-    -> IO (Maybe Stats)
-runNodesForSeconds loglevel baseConf v n (Seconds seconds) write rdb = do
-    stateVar <- newMVar $ emptyConsensusState v
+    -> FilePath
+    -> (forall logger. Int -> StartedChainweb logger -> IO ())
+    -> IO ()
+runNodesForSeconds loglevel write baseConf n (Seconds seconds) rdb pactDbDir inner = do
     void $ timeout (int seconds * 1_000_000)
-        $ runNodes loglevel write stateVar baseConf n rdb
-
-    consensusState <- readMVar stateVar
-    return (consensusStateSummary consensusState)
+        $ runNodes loglevel write baseConf n rdb pactDbDir inner
 
 replayTest
     :: LogLevel
     -> ChainwebVersion
     -> Natural
     -> TestTree
-replayTest loglevel v n = testCaseSteps name $ \step -> do
-    let tastylog = step . T.unpack
-    withTempRocksDb "replay-test" $ \rdb -> do
+replayTest loglevel v n = after AllFinish "ConsensusNetwork" $ testCaseSteps name $ \step -> 
+    withTempRocksDb "replay-test-rocks" $ \rdb -> 
+    withSystemTempDirectory "replay-test-pact" $ \pactDbDir -> do
+        let tastylog = step . T.unpack
         tastylog "phase 1..."
-        Just stats1 <- runNodesForSeconds loglevel (multiConfig v n) v n 60 T.putStrLn rdb
+        stateVar <- newMVar $ emptyConsensusState v
+        let ct = harvestConsensusState (genericLogger loglevel T.putStrLn) stateVar
+        runNodesForSeconds loglevel T.putStrLn (multiConfig v n) n 60 rdb pactDbDir ct
```

Comment:
Do the tests require that `n > 1` here? Could it just be hardcoded to `1` or `2` (if two nodes are required)?

Reply:
`2` should be fine.